homeassistant/components/synology_dsm/__init__.py

```diff
@@ -128,19 +130,33 @@ async def async_coordinator_update_data_cameras() -> dict[
             return None
 
         surveillance_station = api.surveillance_station
+        current_data: dict[str, SynoCamera] = {
+            camera.id: camera for camera in surveillance_station.get_all_cameras()
+        }
 
         try:
             async with async_timeout.timeout(30):
                 await hass.async_add_executor_job(surveillance_station.update)
         except SynologyDSMAPIErrorException as err:
             raise UpdateFailed(f"Error communicating with API: {err}") from err
 
-        return {
-            "cameras": {
-                camera.id: camera for camera in surveillance_station.get_all_cameras()
-            }
+        new_data: dict[str, SynoCamera] = {
+            camera.id: camera for camera in surveillance_station.get_all_cameras()
         }
 
+        for cam_id, cam_data_new in new_data.items():
+            if (
+                (cam_data_current := current_data.get(cam_id)) is not None
+                and cam_data_current.live_view.rtsp != cam_data_new.live_view.rtsp
+            ):
+                async_dispatcher_send(
+                    hass,
+                    f"{SIGNAL_CAMERA_SOURCE_CHANGED}_{cam_id}",
```

Comment:
Is the `cam_id` globally unique?   If not there is a risk of collision if there are multiple synology boxes set up.

Reply:
Looks like it is definitely going to collide if you have multiple
` [synology_dsm.synology_dsm] RESPONSE: {'data': [{'id': 1, `

We need to mix in the config entry entry_id into the signal name

Reply:
you are absolutly right ... will add the config entry id